lib/swiss_village_directory/village.rb

```diff
@@ -1,15 +1,17 @@
+require 'bigdecimal/util'
+
 module SwissVillageDirectory
   class Village
     attr_reader :name, :zip_code, :one_digit_spare, :commune, :canton, :longitude, :latitude
 
     def initialize(name, zip_code, one_digit_spare, commune, canton, longitude, latitude)
       @name = name
-      @zip_code = zip_code.to_i
+      @zip_code = zip_code.to_s
```

Comment:
what where your thoughts about this? i know that a ZIP can be a string, but not in CH, all of them are ints. so we could stay specific?

Reply:
zip codes are not integers:y ou don't sum zip codes or make operations on them. They are strings. This is causing so many troubles in many systems at viseca where they use integers.
Also a zip code might start with zero (not in Switzerland), but I still might search for it. If I need to convert to integer, I might get wrong results. 
In my case for example: they type `08057` which is a zip code in Italy, and I find the zip_Code `8057` in Switzerland.

Reply:
Postleitzahl is an encoded multidimensional label, not a number. So a string is the correct representation.
* There are groups encoded in the first and second digit.
* The corresponding numbers are not evenly spaced.
* You cannot calculate anything, not even distance (e.g. Lausanne-Bulle)